### PR TITLE
dbeaver/dbeaver#17564 hana strip comments before blocks

### LINKED ISSUE #17564: SAP HANA -- Ignore leading comments before anonymous blocks
Strictly speaking, it is a HANA error that occurs here, but dbeaver might circumvent it. Whenever a enclosed comment (/* */) is placed before a anonymous block and included in the statement sent to the HANA db, only statistics from the query are shown as results. Error does not occur when using single line comments (--):

```
/* Issue */
DO BEGIN
	SELECT * FROM dummy;
END;
```
image

```
-- No issue
DO BEGIN
	SELECT * FROM dummy;
END;
```
image

Could the parser be modified as to not automatically include comments before an anonymous block? SAP HANA Web-Based Development Workbench removes the comment, even when manually selecting it, such that the query produces a result. Thanks.

### PR DESCRIPTION
**Pull request by Destrolaric**
Closes dbeaver/dbeaver#17564

### CODE REVIEW

File: test/org.jkiss.dbeaver.test.platform/src/org/jkiss/dbeaver/model/sql/parser/SQLScriptParserTest.java [hunk 1]
```diff
@@ -502,6 +501,41 @@ public void parseCurrentControlCommandsCursorTail() throws DBException {
...
+            Assert.assertEquals("DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END;", element.getText());
+        }
+        {
+            String query = "DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END;";
+            SQLParserContext context = createParserContext(setDialect("sap_hana"), query);
+            SQLScriptElement element = SQLScriptParser.parseQuery(context, 0, query.length(), 0, false, false);
+            Assert.assertEquals("DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END;", element.getText());
+        }
+        {
+            String query = "DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END;";
```
Comment (LonwoLonwo) [inline test/org.jkiss.dbeaver.test.platform/src/org/jkiss/dbeaver/model/sql/parser/SQLScriptParserTest.java]: This test looks exactly like the previous one.

Also, please add more tests not with the Hana dialect.

**Review by E1izabeth — APPROVED** [review]

File: test/org.jkiss.dbeaver.test.platform/src/org/jkiss/dbeaver/model/sql/parser/SQLScriptParserGenericsTest.java [hunk 2]
```diff
@@ -103,6 +99,49 @@ public void parseFromCursorPositionBeginTransaction() throws DBException {
...
+            SQLParserContext context = createParserContext(hanaDialect, query);
+            SQLScriptElement element = SQLScriptParser.parseQuery(context, 0, query.length(), 0, false, false);
+            Assert.assertEquals("DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END", element.getText());
+        }
+        {
+            String query = "DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END;";
+            SQLParserContext context = createParserContext(hanaDialect, query);
+            SQLScriptElement element = SQLScriptParser.parseQuery(context, 0, query.length(), 0, false, false);
+            Assert.assertEquals("DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END", element.getText());
+        }
```
Comment (E1izabeth) [inline test/org.jkiss.dbeaver.test.platform/src/org/jkiss/dbeaver/model/sql/parser/SQLScriptParserGenericsTest.java]: These two cases looks absolutely identical

**Comment by LonwoLonwo** [comment]
Also, please add more tests not with the Hana dialect.

**Review by LonwoLonwo — APPROVED** [review]

**Comment by Matvey16** [comment]
Verified

**Pull request merged**